### ISSUE TITLE
Throws errors when Tern is installed

### ISSUE THREAD

**Issue by spinningarrow** [metakirby5/codi.vim#25]
I don't know if this is an issue with Codi or TernJS, but I often see errors like this while I'm typing:

<img width="856" alt="screen shot 2016-08-19 at 5 37 53 pm" src="https://cloud.githubusercontent.com/assets/1797328/17809620/7b2bfbac-6635-11e6-80fa-ec8d7d9b428a.png">

Any idea what could be causing them and how to fix it?

**Comment by metakirby5** [comment]
That's interesting - how consistently and with what frequency does this issue happen? Is it with every keystroke, or seemingly at random on some keystrokes? My suspicion that the asynchronous buffer switching that Codi does is interrupting the `tern_sendBufferIfDirty` call. In the event where the buffer switches at, for example, [this line of code](https://github.com/ternjs/tern_for_vim/blob/34c516a783b54f2068674ffd581c0f82b8b48bf5/script/tern.py#L240), it's impossible to guarantee correctness, as there is no way for Codi to acquire a communal lock with any other plugin.

**Comment by spinningarrow** [comment]
It's seemingly at random on some keystrokes. Perhaps it's worth disabling Tern when Codi is enabled.

**Comment by metakirby5** [comment]
I'll take a closer look when I get the chance, but it is unlikely that this issue will be fixed. If anything, I'll add a warning about the asynchronous buffer switching Codi does behind the scenes to the documentation.

**Comment by spinningarrow** [comment]
Sure. Thanks for the useful plugin!

**Comment by metakirby5** [comment]
Hey, it just occurred to me - have you tried using Tern + Codi with `g:codi#sync = 1`? This might "fix" the problem, in the sense that it will work with synchronous code.

**Comment by spinningarrow** [comment]
Nope, I haven't tried that yet - let me do that and get back to you.
